### PR TITLE
52 setup access control microservice

### LINKED ISSUE #52: Setup access control microservice
Setup a new microservice that can be used for access control. It should have a database where we can store:
- Who have access to which pipelines
- Who have access to which organisations
- Who have access to which repository
- Who have access to which resource

It should have access methods for resolving if a given user has access to the above e.g. a table `UserPipeline` that has `UserId` and `PipelineId`. If the user has a mapping to the pipeline, then the user is granted access.

This requires all endpoints e.g. when creating a pipeline, then a message should be sent to the new access control microservice that user with id x has created pipeline with id y and then add that as a record to the table `UserPipeline`. 

Now you need to actually add it to the controllers:
- Add to rabbitMQ and setup messages to verify incoming request on controllers in `DAPM.ClientApi`.
- Add e2e tests e.g.:
  - One user creates a pipeline and can get the pipeline.
  - One user creates a pipeline, another user tries to get the pipeline but is denied.

Backend focused - nothing to be added in keycloak.

### LINKED ISSUE #59: Configuration for jenkins
Appsettings should be updated to include keycloak settings for jenkins.

### LINKED ISSUE #85: Move e2e tests out of DAPM backend solution and into own solution

### PR DESCRIPTION
**Pull request by RasmusUK**
Added a new microservice `AccessControlService` to handle access control.
For now it handles access for pipelines, resources and repositories.

When a user creates a pipeline then only that user will have access to the pipeline.
And same for resouraces and repositories.

- All endpoints in clientapi are also requires authorization now.
- Moved e2e tests out of solution into a new solution.
- Updated github workflow to reflect this.
- Created e2e project for the new microservice.
- Created unit tests for it.
- Updated jenkinsfile
- Added appsettings

### CODE REVIEW

File: DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/appsettings.json
Comment (ravvnen) [inline DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/appsettings.json:1]: skal det være localhost her? eller skal det være service navn, siden jeg antager dette bliver kaldt interne fra docker netværkte?

File: DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/Requests/AddUserPipelineRequestMessage.cs
Comment (ravvnen) [inline DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/Requests/AddUserPipelineRequestMessage.cs:1]: Kan se warning her. Evt kig på det

**Review by ravvnen — APPROVED** [review]
solidt arbjede hold da op

File: DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/appsettings.json
Comment (RasmusUK) [reply DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/appsettings.json:1]: End2End testene er flyttet ud til deres eget projekt og er ikke længere en del af docker så det skal være localhost.

File: DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/Requests/AddUserPipelineRequestMessage.cs
Comment (RasmusUK) [reply DAPM.Backend.Test.EndToEnd/DAPM.AccessControlService.Test.EndToEnd/Requests/AddUserPipelineRequestMessage.cs:1]: Det er godkendt når det er dto`er ;)

**Pull request merged**